react-router/react-router.d.ts

```diff
@@ -1,278 +1,171 @@
     }
-    
-    interface HashLocation extends LocationBase, LocationListener {}
-    interface HistoryLocation extends LocationBase, LocationListener {}
-    interface RefreshLocation extends LocationBase {}
-    
-    var HashLocation: HashLocation;
-    var HistoryLocation: HistoryLocation;
-    var RefreshLocation: RefreshLocation;
-    
-    
-    //
-    // Behavior
-    // ----------------------------------------------------------------------
-    interface ScrollBehaviorBase {
-        updateScrollPosition(position: {x: number; y: number;}, actionType: string): void;
-    }
-    interface ImitateBrowserBehavior extends ScrollBehaviorBase {}
-    interface ScrollToTopBehavior extends ScrollBehaviorBase {}
-    
-    var ImitateBrowserBehavior: ImitateBrowserBehavior;
-    var ScrollToTopBehavior: ScrollToTopBehavior;
-    
-    
-    //
-    // Router
-    // ----------------------------------------------------------------------
-    interface Router extends React.ReactElement<any> {
-        run(callback: RouterRunCallback): void;
-    }
-    
-    interface RouterState {
-        path: string;
-        action: string;
-        pathname: string;
-        params: {};
-        query: {};
-        routes : Route[];
-    }
-    
-    interface RouterCreateOption {
-        routes: Route;
-        location?: LocationBase;
-        scrollBehavior?: ScrollBehaviorBase;
-    }
-    
-    type RouterRunCallback = (Handler: Router, state: RouterState) => void;
-    
-    function create(options: RouterCreateOption): Router;
-    function run(routes: Route, callback: RouterRunCallback): Router;
-    function run(routes: Route, location: LocationBase, callback: RouterRunCallback): Router;
-    
-    
-    //
-    // History
-    // ----------------------------------------------------------------------
-    interface History {
-        back(): void;
-        length: number;
+
+    interface MutableLocation {
+        push(path: string): void;
+        replace(path: string): void;
+        pop(): void;
     }
-    var History: History;
-    
-    
-    //
-    // Transition
-    // ----------------------------------------------------------------------
+
+    interface Location extends LocationBase, LocationListener, MutableLocation {}
+    interface RefreshLocation extends LocationBase, MutableLocation {}
+
+    var HashLocation:    Location;
+    var HistoryLocation: Location;
+    var RefreshLocation: RefreshLocation;
+    var LocationBase:    LocationBase;
+    var TestLocation:    Location;
+
     interface Transition {
         abort(): void;
-        redirect(to: string, params?: {}, query?: {}): void;
+        redirect(to: string, params?: any, query?: any): void;
         retry(): void;
     }
-    
-    interface TransitionStaticLifecycle {
-        willTransitionTo?(
+
+    //
+    // Renderable Components
+    //
+    interface RouteHandler extends React.ComponentClass<any> {
+        willTransitionTo(
```

Comment:
Is this right? I don't see `willTransitionTo` or `willTransitionFrom` in https://github.com/rackt/react-router/blob/master/modules/components/RouteHandler.js